Classes/Controller/Haiku/AjaxController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace T3docs\Examples\Controller\Haiku;
+
+use Psr\Http\Message\ServerRequestInterface;
+use T3docs\Examples\Utility\HaikuSeasonUtility;
+use TYPO3\CMS\Core\Http\PropagateResponseException;
+use TYPO3\CMS\Core\Localization\LanguageService;
+use TYPO3\CMS\Core\Localization\LanguageServiceFactory;
+
+class AjaxController
+{
+    public array $conf = [];
+    private LanguageService $languageService;
+
+    public function __construct(
+        private readonly LanguageServiceFactory $languageServiceFactory,
+    ) {
+    }
+
+    private function getLanguageService(ServerRequestInterface $request): LanguageService
+    {
+        return $this->languageServiceFactory->createFromSiteLanguage(
+            $request->getAttribute('language')
+            ?? $request->getAttribute('site')->getDefaultLanguage()
+        );
+    }
+
+    public function main(string $content, array $conf, ServerRequestInterface $request): string
+    {
+        $this->conf = $conf;
+        $this->languageService = $this->getLanguageService($request);
+
+        $parameter = $request->getQueryParams()['tx_examples_haiku']??[];
```

Comment:
```suggestion
        $parameter = $request->getQueryParams()['tx_examples_haiku'] ?? [];
```

Reply:
Can we adjust the CI so it catches this? It is kind of stupid to always change such things manually?

Reply:
Yup, is on my todo.